serve/mlc_serve/engine/engine_common.py

```diff
@@ -352,6 +352,7 @@ def evict_request(self, cancell_callback: Callable[[RequestId], None]) -> int:
                 continue
 
             self.remove_request_from_batch(request_to_remove.request_id)
+            request_to_remove.generation_sequences[0].next_start_position = 0
```

Comment:
I think the convention of using `next_start_position == 0` to represent "a request that hasn't been prefilled" is confusing and should be removed.

* Add `is_prefilled` field to `RequestState`
* Set `is_prefilled` to False here
* Check `is_prefilled` instead of `next_start_position` at https://github.com/octoml/mlc-llm/blob/batch-serving/serve/mlc_serve/engine/engine_common.py#L192 and everywhere else that does `next_start_position == 0` check

And `next_start_position` usage should be updated to
* `next_start_position=len(prompt_tokens)` at https://github.com/octoml/mlc-llm/blob/batch-serving/serve/mlc_serve/engine/engine_common.py#L52 
etc.

Reply:
i agree and I do not fully understand the meaning of `the next_start_position`. It is used in the calculation of number of tokens for decode, but logic is very indirect. By fact the number of new calculated decod tokens depend on the number of previously decoded. It should be streamlined.

Reply:
Yeah, indeed, I ran into `next_start_position` when I was working on detokenization and this seemed unnecessary at glance.  Although I need to take a deeper look, but there are seemingly a couple of ways to represent the same info with different ways.  I'd like us to clarify this and clearly document so that we all have a good understanding how these work.